bot.py

```diff
@@ -0,0 +1,300 @@
+    """Достаёт из файла setup.cfg перечень id учётных запписей телеграм."""
+    logger.info('***Работает get_subscribers_ids.')
+    telegram_subscriber_ids = []
+    tg_ids = config['tg_setting']['subscription_tg_ids'].split(',')
+    for tg_id in tg_ids:
+        if isinstance(int(tg_id), int):
+            telegram_subscriber_ids.append(int(tg_id))
+    logger.debug(
+        'Список ИД учёток Телеграм подписчиков из get_subscribers_ids - '
+        + f'{telegram_subscriber_ids}.')
+    return telegram_subscriber_ids
+
+
+def send_message_admin(bot, message):
+    """Отправляет сообщение в Telegram чат админа."""
+    logger.info('***Работает send_message_admin.')
+    if not bot:
+        message = f'Ошибка инициализации объекта bot - {bot}.'
+        logger.error(message, exc_info=True)
+        raise MyCustomError(message)
+    else:
+        bot.send_message(chat_id=TELEGRAM_ADMIN_ID, text=message,)
+        logger.info(f'Админу отправлено сообщение - "{message}".')
+
+
+def send_message(bot, message, telegram_ids):
+    """Отправляет сообщение в Telegram-чаты участников процесса."""
+    logger.info('***Работает send_message.')
+    if not bot:
+        message = f'Ошибка инициализации объекта bot - {bot}.'
+        logger.error(message, exc_info=True)
+        raise MyCustomError(message)
+    else:
+        for t in telegram_ids:
+            bot.send_message(chat_id=t, text=message,)
+        logger.info(f'Получателям отправлено сообщение - "{message}".')
+
+
+def check_status_resource(bot, endpoint, telegram_ids):
+    """Делает запрос к эндпоинту.
+
+    В качестве параметра функция получает временную метку и ендпоинт. Делает
+    запрос и, если статус ответа не 200, то посылает сообщение пользоватлю из
+    списка.
+    """
+    logger.info('***Работает check_status_resource.')
+    response = requests.get(endpoint)
+    logger.info(f'response - "{response}". type(response) - {type(response)}.')
+    status_code = response.status_code
+    if response.status_code != HTTPStatus.OK:
+        message_status_code_not_200 = (
+            f'Ошибка запроса к сайту "{endpoint}". Код статуса - '
+            + f'{status_code}.')
+        logger.error(message_status_code_not_200)
+        send_message(
+            bot, message_status_code_not_200, telegram_ids)
+    else:
+        logger.info(f'Сайтттт - "{endpoint}". Код статуса - {status_code}.')
+    return status_code
+
+
+def check_tokens():
+    """Проверяет доступность переменных окружения для работы программы."""
+    logger.info('***Работает check_tokens.')
+    logger.debug(f'TELEGRAM_TOKEN - {TELEGRAM_TOKEN}.')
+    logger.debug(f'TELEGRAM_ADMIN_ID - {TELEGRAM_ADMIN_ID}.')
+    return all([TELEGRAM_TOKEN, TELEGRAM_ADMIN_ID])
+
+
+def last_check(update, context):
+    """Возвращает результаты последней проверки."""
+    logger.info('***Работает last_check.')
+    chat = update.effective_chat
+    name = update.message.chat.username
+    message = ''.join(CHECK_RESULT)
```

Comment:
а нужен ли здесь .join? он же ничего не делает. Может просто
```suggestion
    message = CHECK_RESULT
```
или вообще избавиться от лишней переменной

Reply:
Согласен. Убрал join.

Reply:
Нет, вернул. Без этого список не конвертируется в строку и некорректно применяется кодировка.